frontend/src/components/Header.js

```diff
@@ -1,10 +1,32 @@
 "use client";
 
-import { useState } from "react";
+import { useState, useEffect } from "react";
 import Link from "next/link";
+import { tokenService } from "../utils/auth";
+import LogoutButton from "./LogoutButton";
 
 export default function Header() {
   const [isMenuOpen, setIsMenuOpen] = useState(false);
+  const [user, setUser] = useState(null);
+
+  useEffect(() => {
+    // Get user data from localStorage
+    const userData = tokenService.getUserData();
+    const token = tokenService.getToken();
+    
+    if (token && userData) {
+      setUser(userData);
+    }
+  }, []);
+
+  const getUserInitials = (name) => {
```

Comment:
There are some edge cases that aren't covered here:
- `getUserInitials(" John Doe")` will return `" J"`, consider trimming the input with a `.trim()` function
- `getUserInitials("John   Doe")` will return `"J "`, consider adding a split function for whitespace `.split(/\s+/)`

`const nameParts = name.trim().split(/\s+/);` would probably be 'safer'